### PR TITLE
Makes dartboards functional

### PR DESCRIPTION
**Pull request by VileBeggar**
# About the pull request

adds basic functionality to dartboards and makes them craftable with cardboard.
fixes a bug where some signs lose their icon when deconstructed.

# Explain why it's good for the game
the dartboard's been sitting there, looking pretty and doing nothing else for a good while. more intractability with props is always fun to see.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

https://github.com/cmss13-devs/cmss13/assets/17518895/4496198d-5a7d-4511-a5bf-894c4df98588

</details>

# Changelog
:cl:
add: Dartboards are now functional and can be crafted with cardboard.
fix: Fixed an issue with some deconstructed signs losing their icon.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/misc.dm [hunk 1]
```diff
@@ -189,3 +189,123 @@
...
+	name = "dartboard"
+	desc = "A dartboard, loosely secured."
+	icon = 'icons/obj/structures/props/props.dmi'
+	icon_state = "dart_board"
+	density = TRUE
+	unslashable = TRUE
+
+/obj/structure/dartboard/get_examine_text()
+	. = ..()
+	if(contents.len)
```
Comment (harryob) [inline code/game/objects/structures/misc.dm:207]:
```suggestion
	if(length(contents))
```

File: code/game/objects/structures/misc.dm [hunk 2]
```diff
@@ -189,3 +189,123 @@
...
+		var/is_are = "is"
+		if(contents.len != 1)
```
Comment (harryob) [inline code/game/objects/structures/misc.dm:209]:
```suggestion
		if(length(contents) != 1)
```

File: code/game/objects/structures/misc.dm [hunk 3]
```diff
@@ -189,3 +189,123 @@
...
+	if(pass_flags)
+		pass_flags.flags_can_pass_all = PASS_MOB_IS
+
+/obj/structure/dartboard/get_projectile_hit_boolean(obj/projectile/projectile)
+	. = ..()
+	visible_message(SPAN_DANGER("[projectile] hits [src], collapsing it!"))
+	collapse()
+
+/obj/structure/dartboard/proc/flush_contents()
+	for(var/atom/movable/embedded_items in contents)
```
Comment (harryob) [inline code/game/objects/structures/misc.dm:225]:
```suggestion
	for(var/atom/movable/embedded_items as anything in contents)
```
contents can only have /atom/movables in it, so we can skip the typecheck here

File: code/game/objects/structures/misc.dm [hunk 4]
```diff
@@ -189,3 +189,123 @@
...
+	unslashable = TRUE
+
+/obj/structure/dartboard/get_examine_text()
+	. = ..()
+	if(contents.len)
+		var/is_are = "is"
+		if(contents.len != 1)
+			is_are = "are"
+
+		. += SPAN_NOTICE("There [is_are] [contents.len] item\s embedded into [src].")
```
Comment (harryob) [inline code/game/objects/structures/misc.dm:212]:
```suggestion
		. += SPAN_NOTICE("There [is_are] [length(contents)] item\s embedded into [src].")
```

File: code/game/objects/structures/misc.dm [hunk 5]
```diff
@@ -189,3 +189,123 @@
...
+	for(var/atom/movable/embedded_items in contents)
+		embedded_items.forceMove(loc)
+
+/obj/structure/dartboard/proc/collapse()
+	playsound(src, 'sound/effects/thud1.ogg', 50)
+	new /obj/item/dartboard/(loc)
+	qdel(src)
+
+/obj/structure/dartboard/attack_hand(mob/user)
+	if(contents.len)
```
Comment (harryob) [inline code/game/objects/structures/misc.dm:234]:
```suggestion
	if(length(contents))
```

File: code/game/objects/structures/misc.dm [hunk 6]
```diff
@@ -189,3 +189,123 @@
...
+
+/obj/structure/dartboard/MouseDrop(over_object, src_location, over_location)
+	. = ..()
+	if(over_object == usr && Adjacent(usr))
+		if(!ishuman(usr))
+			return
+		visible_message(SPAN_NOTICE("[usr] unsecures [src]."))
+		var/obj/item/dartboard/unsecured_board = new(loc)
+		usr.put_in_hands(unsecured_board)
+		qdel(src)
```
Comment (harryob) [inline code/game/objects/structures/misc.dm:280]:
```suggestion
	if(over_object != usr || !Adjacent(usr))
		return
		
	if(!ishuman(usr))
		return

	visible_message(SPAN_NOTICE("[usr] unsecures [src]."))
	var/obj/item/dartboard/unsecured_board = new(loc)
	usr.put_in_hands(unsecured_board)
	qdel(src)
```

**Review by harryob — CHANGES_REQUESTED** [review]
very nice stuff all round, just a few nits

File: code/game/objects/structures/misc.dm (hunk 1 above)
Comment (harryob) [reply code/game/objects/structures/misc.dm:207]: i know it looks more stupid, but we prefer length() to .len as it's faster (by an absurdly small margin) and won't cause runtimes on null (but contents'll never be null sooo 🤷‍♂️)

**Review by harryob — APPROVED** [review]

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by harryob — APPROVED** [review]
i need to free you from map conflict hell

**Pull request merged**